presto/presto.go

```diff
@@ -751,12 +751,13 @@ func (qr *driverRows) Next(dest []driver.Value) error {
 	if qr.columns == nil || qr.rowindex >= len(qr.data) {
 		if qr.nextURI == "" {
 			qr.err = io.EOF
+			return &EOF{QueryID: qr.id}
 		}
 		if err := qr.fetch(true); err != nil {
 			qr.err = err
-		}
-		if qr.err == io.EOF {
-			return &EOF{QueryID: qr.id}
+			if qr.err == io.EOF {
+				return &EOF{QueryID: qr.id}
+			}
```

Comment:
I think an additional `return qa.err` is needed here, otherwise the error is ignored and then you hit the index out of range again. (after the check for EOF).

Reply:
SGTM; @hasan-dot wdyt?